### PR TITLE
Support Pinot Hexed BYTES of PinotDataTableImplv4

### PR DESCRIPTION
**Pull request by INNOCENT-BOY**
You can't query read bytes column with hex encoded. When pinot enabled datatablev4. It will report below error.
image

### CODE REVIEW

File: plugin/trino-pinot/src/main/java/io/trino/plugin/pinot/PinotSegmentPageSource.java [hunk 1]
```diff
@@ -347,7 +347,7 @@ private Slice getSlice(int rowIndex, int columnIndex)
             return getUtf8Slice(field);
         }
         if (trinoType instanceof VarbinaryType) {
-            return Slices.wrappedBuffer(toBytes(currentDataTable.getDataTable().getString(rowIndex, columnIndex)));
+            return Slices.wrappedBuffer(toBytes(currentDataTable.getDataTable().getBytes(rowIndex, columnIndex)));
```
Comment (ebyhr) [inline plugin/trino-pinot/src/main/java/io/trino/plugin/pinot/PinotSegmentPageSource.java:350]: Please add a test case. The test should fail without change.

File: plugin/trino-pinot/src/main/java/io/trino/plugin/pinot/decoders/VarbinaryDecoder.java [hunk 2]
```diff
@@ -45,13 +46,16 @@ else if (value instanceof String) {
         }
     }
 
-    public static byte[] toBytes(String stringValue)
+    public static byte[] toBytes(Object value)
```
Comment (ebyhr) [inline plugin/trino-pinot/src/main/java/io/trino/plugin/pinot/decoders/VarbinaryDecoder.java:49]: `ByteArray` argument is used only from `PinotSegmentPageSource`. Revert this change and inline `Hex.decodeHex`.

**Comment by ebyhr** [comment]
Please don't remove a PR template.

File: plugin/trino-pinot/src/main/java/io/trino/plugin/pinot/PinotSegmentPageSource.java (hunk 1 above)
Comment (INNOCENT-BOY) [reply plugin/trino-pinot/src/main/java/io/trino/plugin/pinot/PinotSegmentPageSource.java:350]: I don't think so. The root cause is that before DataTablev4，DataTable V2/V3 uses String to store BYTES value. So you will get String via fetching bytes columns. But when you enable DataTableV4, you can't use getString to visist bytes column that is store bytes other than string.
And This modification is also compatible with DataTable V2/V3., because class DataTableV2/DataTableV3 don't implement getString or getBytes function, both of those all use BaseDataTable that is trino class. So this modification won't affect legacy DataTable.
At last, I have test different Pinot Cluster that use DataTableV3 or DataTableV4. All query works well.
If you must ask me to add a test case, please provide a test class that test legacy DataTableV3. And I will reference it.

File: plugin/trino-pinot/src/main/java/io/trino/plugin/pinot/decoders/VarbinaryDecoder.java (hunk 2 above)
Comment (INNOCENT-BOY) [reply plugin/trino-pinot/src/main/java/io/trino/plugin/pinot/decoders/VarbinaryDecoder.java:49]: Good suggestion. I will change according your advice.

**Comment by ebyhr** [comment]
Please remove unrelated commits. https://github.com/trinodb/trino/pull/19025/commits
